tests/test_visualize.py

```diff
@@ -17,14 +17,13 @@
 reference_file = DIR_TESTDATA / "test_example_report.pdf"
 
 
-def test_generate_pdf_main():
+def test_generate_pdf_example():
+    """Test generation of example visualization report."""
     # Delete the old generated file if it exists
     if os.path.isfile(output_file):
         os.remove(output_file)
 
-    # Execute the __main__ block of visualize.py by running it as a script
-    script_path = Path(__file__).parent.parent / "src" / "akkudoktoreos" / "utils" / "visualize.py"
-    subprocess.run(["python", str(script_path)], check=True)
+    generate_example_report(filename)
```

Comment:
from class:
# If self.filename is already a valid path, use it; otherwise, combine it with output_dir
so a path to the test folder could generate it there and not in the output folder.

Reply:
will be adapted in #257